apps/restconf/restconf_methods.c

```diff
@@ -581,6 +581,367 @@ api_data_write(clicon_handle h,
 } /* api_data_write */
 
 #ifdef YANG_PATCH
+
+char * init_str() 
+{
+    char* s;
+    s = malloc(TEMP_STR_MALLOC_SIZE);
+    memset(s, 0, TEMP_STR_MALLOC_SIZE);
+    return s;
+}
+
+int cpy_str(char *dest, char *src, size_t size)
+{
+    if (src == NULL) {
+        return 0;
+    }
+    if (dest == NULL) {
+        init_str(dest);
+    }
+    if (size <= 0 || size > TEMP_STR_MALLOC_SIZE) {
+        return 0;
+    }
+    size_t i;
+    for (i = 0; i < size - 1 && src[i]; i++) {
+        dest[i] = src[i];
+    }
+    dest[i] = '\0';
+    return i;
+}
+
+int cat_str(char *dest, char *src, size_t size)
+{
+    if (src == NULL) {
+        return 0;
+    }
+    if (dest == NULL) {
+        init_str(dest);
+    }
+    if (size <= 0 || size > TEMP_STR_MALLOC_SIZE) {
+        return 0;
+    }
+    size_t i;
+    int old_len = strlen(dest);
+    for (i = 0; i < size - 1 && src[i]; i++) {
+        dest[i + old_len] = src[i];
+    }
+    dest[i + old_len] = '\0';
+    return i;
+}
+
+void free_mem(void *str)
+{
+    if (str != NULL)
+        free(str);
+}
+
+int get_xval(
+    cvec* nsc,
+    cxobj* xn,
+    char* val,
+    cxobj **vec,
+    size_t veclen,
+    const char* key
+    )
+{
+    char* tmp_val = NULL;
+    int ret = xpath_vec(xn, nsc, "%s", &vec, &veclen, key);
+    if (ret < 0) {
+        return ret;
+    }
+    for (int j = 0; j < veclen; j++) {
+        cxobj *xn = vec[j];
+        tmp_val = xml_body(xn);
+    }
+    cpy_str(val, tmp_val, TEMP_STR_MALLOC_SIZE);
+    return 0;
+}
+
+int do_replace (
```

Comment:
We can have another function called do_replace. I suggest to either make it static, or rename it.